modules/platform/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/providers/OAuth2ServiceProviderRegistryImpl.java

```diff
@@ -21,65 +21,68 @@
 import java.io.Serializable;
 import java.util.Arrays;
 import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 import org.apache.commons.lang3.StringUtils;
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.nuxeo.ecm.core.api.DocumentModel;
 import org.nuxeo.ecm.directory.BaseSession;
 import org.nuxeo.ecm.directory.DirectoryException;
 import org.nuxeo.ecm.directory.Session;
 import org.nuxeo.ecm.directory.api.DirectoryService;
 import org.nuxeo.runtime.api.Framework;
 import org.nuxeo.runtime.model.ComponentContext;
-import org.nuxeo.runtime.model.ComponentInstance;
 import org.nuxeo.runtime.model.DefaultComponent;
 
 /**
  * Implementation of the {@link OAuth2ServiceProviderRegistry}. The storage backend is a SQL Directory.
  */
 public class OAuth2ServiceProviderRegistryImpl extends DefaultComponent implements OAuth2ServiceProviderRegistry {
 
-    protected static final Log log = LogFactory.getLog(OAuth2ServiceProviderRegistryImpl.class);
+    private static final Logger log = LogManager.getLogger(OAuth2ServiceProviderRegistryImpl.class);
 
     public static final String PROVIDER_EP = "providers";
 
     public static final String DIRECTORY_NAME = "oauth2ServiceProviders";
 
     public static final String SCHEMA = "oauth2ServiceProvider";
 
-    /**
-     * Registry of contributed providers. These providers can extend and/or override the default provider class.
-     */
-    protected OAuth2ServiceProviderContributionRegistry registry = new OAuth2ServiceProviderContributionRegistry();
-
-    protected DocumentModel getProviderDocModel(String serviceName) {
-        try {
-            if (StringUtils.isBlank(serviceName)) {
-                log.warn("Can not find provider without a serviceName!");
-                return null;
-            }
+    protected static final String SERVICE_NAME_KEY = "serviceName";
 
-            Map<String, Serializable> filter = new HashMap<>();
-            filter.put("serviceName", serviceName);
+    @Override
+    public void start(ComponentContext context) {
+        persistProviders();
+    }
 
-            List<DocumentModel> providers = queryProviders(filter, 1);
-            return providers.isEmpty() ? null : providers.get(0);
-        } catch (DirectoryException e) {
-            log.error("Unable to read provider from Directory backend", e);
+    protected DocumentModel getPersistedProvider(String serviceName) {
+        if (StringUtils.isBlank(serviceName)) {
+            log.warn("Cannot fetch provider without a serviceName");
```

Comment:
Maybe, but there are multiple places where the null provider is checked: sometimes null is indeed expected (sometimes not), I preserved the existing behavior to avoid getting into too much cleanup, please tell if you'd rather have me do this cleanup (which would avoid getting a NPE only in jsf-ui code).

Reply:
We can do other cleanups later